hil/ext/switches/brocade.py

```diff
@@ -298,8 +299,8 @@ def _construct_tag(name):
 
     def _make_request(self, method, url, data=None,
                       acceptable_error_codes=()):
-        r = requests.request(method, url, data=data, auth=self._auth)
-        if r.status_code >= 400 and \
-           r.status_code not in acceptable_error_codes:
-            logger.error('Bad Request to switch. Response: %s', r.text)
+        try:
+            r = requests.request(method, url, data=data, auth=self._auth)
+        except SwitchError:
```

Comment:
@naved001 ping. Not sure whether this is the correct approach of handling the error. If not, would you mind telling me what would be the correct way?

For now, I am not sure whether `requests.request()` here will raise a `SwitchError`. Is there a way to test this? 

Thanks!